ods_ci/tests/Tests/0100__platform/0101__deploy/0104__operators/0105__serverless_operator/0105__serverless_operator.robot

```diff
@@ -5,45 +5,137 @@ Library         OpenShiftLibrary
-    Resource Should Exist     Gateway    kserve-local-gateway     ${ISTIO_NS}
-    Resource Should Exist     Service    kserve-local-gateway     ${ISTIO_NS}
-    Resource Should Exist     Service    knative-local-gateway     ${ISTIO_NS}
-    Resource Should Exist     deployment    controller     ${KNATIVESERVING_NS}
+    ...    Resource Should Exist     KnativeServing    knative-serving     ${KNATIVESERVING_NS}
+
+    Wait Knative Serving CR To Be In Ready State
+
+    Wait Until Keyword Succeeds    2 min    0 sec
+    ...    Resource Should Exist     Gateway    knative-ingress-gateway     ${KNATIVESERVING_NS}
+
+    Wait Until Keyword Succeeds    2 min    0 sec
+    ...    Resource Should Exist     Gateway    knative-local-gateway     ${KNATIVESERVING_NS}
+
+    Wait Until Keyword Succeeds    2 min    0 sec
+    ...    Resource Should Exist     Gateway    kserve-local-gateway     ${ISTIO_NS}
+
+    Wait Until Keyword Succeeds    2 min    0 sec
+    ...    Resource Should Exist     Service    kserve-local-gateway     ${ISTIO_NS}
+
+    Wait Until Keyword Succeeds    2 min    0 sec
+    ...    Resource Should Exist     Service    knative-local-gateway     ${ISTIO_NS}
+
+    Wait Until Keyword Succeeds    2 min    0 sec
+    ...    Resource Should Exist     deployment    controller     ${KNATIVESERVING_NS}
+
     Wait For Pods Numbers  2    namespace=${KNATIVESERVING_NS}
-    ...    label_selector=app.kubernetes.io/component=controller    timeout=120
-    ${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
-    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
+    ...    label_selector=${KNATIVE_SERVING_CONTROLLER_LABEL_SELECTOR}    timeout=120
+
+    #${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    ${KNATIVE_SERVING_CONTROLLER_LABEL_SELECTOR}    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
     #${podname}=    Get Pod Name   ${OPERATOR_NAMESPACE}    ${OPERATOR_LABEL_SELECTOR}
     #Verify Pod Logs Do Not Contain    ${podname}    ${OPERATOR_NAMESPACE}    ${regex_pattern}    rhods-operator
-    Wait For DSC Conditions Reconciled    ${KNATIVESERVING_NS}    default-dsc
+
+Validate DSC Knative Serving Removed State
+    [Documentation]    Validate that KServe Serving state Removed does remove relevant resources.
+    [Tags]  Operator    RHOAIENG-7217    Tier2    kserve-serving-removed
+
+    ${management_state}=    Get DSC Component State    ${DSC_NAME}    kserve/serving    ${OPERATOR_NAMESPACE}
+    IF    "${management_state}" != "Removed"
+            Set Component State    kserve/serving    Removed
+    END
+
+    Wait Until Keyword Succeeds    5 min    0 sec
+    ...    Is Resource Present     KnativeServing    knative-serving    ${KNATIVESERVING_NS}   ${IS_NOT_PRESENT}
+
+    Wait For DSC Conditions Reconciled    ${OPERATOR_NAMESPACE}    default-dsc
+
+    Wait Until Keyword Succeeds    2 min    0 sec
```

Comment:
I would set 5 minutes as the timeout for all the keywords.

Reply:
Changed timeouts to 5 min.